Keas.Mvc/Controllers/Api/AccessController.cs

```diff
@@ -130,15 +130,38 @@ public async Task<IActionResult> Delete([FromBody]Access access)
                 return BadRequest(ModelState);
             }
 
-            if(!access.Active || access.Assignments.Count > 0) // assignments should have been revoked
+            if(!access.Active) 
             {
                 return BadRequest(ModelState);
             }
-            _context.Access.Update(access);
-            access.Active = false;
-            await _context.SaveChangesAsync();
-            // TODO: track history?
-            return Json(null);
+
+            using(var transaction = _context.Database.BeginTransaction())
+            {
+                try
+                {
+                    _context.Access.Update(access);
+
+                    if(access.Assignments.Count > 0)
+                    {
+                        foreach(var assignment in access.Assignments.ToList()) 
+                        {
+                            await _eventService.TrackUnAssignAccess(assignment, Team); // call before we remove person info
```

Comment:
Not a huge issue but under the covers this tracking does a saveChanges() which is probably resulting in just tons of DB calls that are unnecessary.  If it becomes a problem we should split out and let the caller handle the persistence